Microsoft.Toolkit.Uwp.UI.Controls/DropShadowPanel/DropShadowPanel.cs

```diff
@@ -73,141 +62,34 @@ public DropShadowPanel()
-            }
-        }
-
-        /// <summary>
-        /// Gets DropShadow. Exposes the underlying composition object to allow custom Windows.UI.Composition animations.
-        /// </summary>
-        public DropShadow DropShadow => _dropShadow;
-
-        /// <summary>
-        /// Gets or sets the mask of the underlying <see cref="Windows.UI.Composition.DropShadow"/>.
-        /// Allows for a custom <see cref="Windows.UI.Composition.CompositionBrush"/> to be set.
-        /// </summary>
-        public CompositionBrush Mask
-        {
-            get
-            {
-                return _dropShadow.Mask;
-            }
-
-            set
-            {
-                _dropShadow.Mask = value;
-            }
-        }
-
-        private void CompositionShadow_SizeChanged(object sender, SizeChangedEventArgs e)
-        {
-            UpdateShadowSize();
-        }
-
-        private void ConfigureShadowVisualForCastingElement()
-        {
-            UpdateShadowMask();
-
-            UpdateShadowSize();
-        }
-
-        private void OnBlurRadiusChanged(double newValue)
-        {
-            _dropShadow.BlurRadius = (float)newValue;
-        }
-
-        private void OnColorChanged(Color newValue)
-        {
-            _dropShadow.Color = newValue;
-        }
-
-        private void OnOffsetXChanged(double newValue)
-        {
-            UpdateShadowOffset((float)newValue, _dropShadow.Offset.Y, _dropShadow.Offset.Z);
-        }
-
-        private void OnOffsetYChanged(double newValue)
-        {
-            UpdateShadowOffset(_dropShadow.Offset.X, (float)newValue, _dropShadow.Offset.Z);
-        }
-
-        private void OnOffsetZChanged(double newValue)
-        {
-            UpdateShadowOffset(_dropShadow.Offset.X, _dropShadow.Offset.Y, (float)newValue);
-        }
-
-        private void OnShadowOpacityChanged(double newValue)
-        {
-            _dropShadow.Opacity = (float)newValue;
-        }
+                Source = this,
+                Path = new PropertyPath("CastingElement"),
+                Mode = BindingMode.OneWay
+            };
 
-        private void UpdateShadowMask()
-        {
-            if (IsSupported)
-            {
-                if (_contentElement != null)
-                {
-                    CompositionBrush mask = null;
-                    if (_contentElement is Image)
-                    {
-                        mask = ((Image)_contentElement).GetAlphaMask();
-                    }
-                    else if (_contentElement is Shape)
-                    {
-                        mask = ((Shape)_contentElement).GetAlphaMask();
-                    }
-                    else if (_contentElement is TextBlock)
-                    {
-                        mask = ((TextBlock)_contentElement).GetAlphaMask();
-                    }
+            _contentPresenter.SetBinding(ContentPresenter.ContentProperty, contentBinding);
```

Comment:
Setting this up as a binding feels hokey. This could be accomplished all through templating (and might not even be needed by inheriting from ContentControl and thus CastingElement can be eliminated)